samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -429,10 +428,47 @@ static int fds_init(struct mqtt_client *c)
 	return 0;
 }
 
+static void button_handler(uint32_t button_states, uint32_t has_changed)
+{
+	uint8_t btn_num;
+	uint8_t btn_state;
+	int ret;
+
+	while (has_changed) {
```

Comment:
I don't see a reason for this loop if we're interested in a particular button, i.e. why not simply
```
if (has_changed & button_states & BIT(CONFIG_BUTTON_EVENT_BTN_NUM - 1)) {
        /* your button press logic */
}
```